src/clients/sidecar/apis/ConnectionsResourceApi.ts

```diff
@@ -39,6 +39,7 @@ export interface GatewayV1ConnectionsIdPutRequest {
 }
 
 export interface GatewayV1ConnectionsPostRequest {
+  dry_run?: boolean;
```

Comment:
@Cerchie looks like that "test connection" is even more closely tied to Submit action than I thought! 
In this case `topic-config-form.ts` could be inspiration for implementing this type of thing since it also has a dry run option that we run first (though I'm sure it can be improved upon)